### ISSUE TITLE
Incorrect repo name and unhelpful error messages in 'grunt build' string processing

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/chipper#1014]
Related to [this Google Group post](https://groups.google.com/g/developing-interactive-simulations-in-html5/c/cS8WqHpBcBs), 

```
% cd perennial
% grunt create-sim --repo=Numerical-Differentiation --author=PixelZoom --title="Numerical Differentiation"
// add NumericalDifferentiationStrings.custom somewhere in the .js code
% cd Numerical-Differentiation 
% grunt build
...
Fatal error: Perennial task failed:
AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: Missing string information for numerical-differentiation custom
```

Problems:

(1) The repo is named Numerical-Differentiation and the error message says "numerical-differentiation". Something is assuming lowercase repo names, and doing an inappropriate conversion.  I'm not sure if this causes any problems other than an incorrect error message.

(2) The error message is cryptic, not at all useful.  For a new sim developer, this error message is not at all helpful. Based on my experience, I can guess that "numerical-differentiation" is the basename of the strings file, and "custom" is the string key, and then infer that this key is missing from the string file.    Please make the error message describe the problem clearly, and (ideally) how to fix it.

**Comment by jonathanolson** [comment]
Our toolchain makes assumptions about file structure currently, and there is an assumption that repo names are "lower-case letters with dashes". Does it sound acceptable to document that and fail out in `grunt create-sim`?

**Comment by pixelzoom** [comment]
If I were a 3rd-party developer, and it was my project that suddenly didn't work with PhET build tools, then my answer to your question would be "no, that's not acceptable".   "Numerical-Differentiation" is not the first example I've come across that doesn't follow this implicit convention.  I've encountered other projects in the Google Group that don't follow that convention.  So build tools are being used outside of PhET, and if you make this convention explicit, then you're going to impact some 3rd-party developers.  My suspicion is that this convention isn't necessary, but instead became a convention due to some implementation decisions. And if a convention was necessary, this problem could have been avoided by proper use of assertions to make the assumption explicit.   But here we are.  So... up to you.

**Comment by jonathanolson** [comment]
I'm fine getting things set up so it can more explicitly be used in other ways.

**Comment by jonathanolson** [comment]
`grunt create-sim --repo="A Test Repo" --author=Author --title="Some Test Title"` is failing out with other errors, so it looks like I'll want to patch up create-sim first.

**Comment by jonathanolson** [comment]
I've spent a good chunk of time on this, but it seems that `npm install` fails and expects specific naming conventions for package names in https://docs.npmjs.com/cli/v7/configuring-npm/package-json, specifically:

> The name must be less than or equal to 214 characters. This includes the scope for scoped packages.
The names of scoped packages can begin with a dot or an underscore. This is not permitted without a scope.
New packages must not have uppercase letters in the name.
The name ends up being part of a URL, an argument on the command line, and a folder name. Therefore, the name can't contain any non-URL-safe characters.

This results in `npm install` being a no-op, and that sounds difficult to work around. Additionally, casing requirements make string key determinations incredibly difficult, and spaces throw all sorts of wrenches (things like CWDs need escaping, other things don't).

Since we're committed to using the npm system (and things fail when we go outside of this due to external dependencies), I believe it's safe to stick to these. I've updated documentation in our new-repo checklist.

@pixelzoom objections? Other places to document?
